src/tir/ir/expr.cc

```diff
@@ -707,7 +707,8 @@ Ramp::Ramp(PrimExpr base, PrimExpr stride, int lanes, Span span) {
   ICHECK(base.dtype().is_scalar());
   ICHECK(stride.dtype().is_scalar());
   ICHECK_GT(lanes, 1);
-  ICHECK_EQ(stride.dtype(), base.dtype());
+  ICHECK(base.dtype().is_int());
+  ICHECK(stride.dtype().is_int());
```

Comment:
After the change, the ramp node will produce a checking error since the shape variable is now of `int64` while other operands may be of `int32`.